opm.Dockerfile

```diff
@@ -0,0 +1,8 @@
+# NOTE: This Dockerfile is used in conjuction with GoReleaser to
+#   build opm images. See the configurations in .goreleaser.yaml
+#   and .github/workflows/release.yaml.
+
+FROM gcr.io/distroless/base:debug
+COPY ["nsswitch.conf", "/etc/nsswitch.conf"]
+COPY opm /bin/opm
```

Comment:
I'm not super familiar with the current role of the upstream-opm-builder.Dockerfile - it looks like we're blindly copying a local version of opm without building it, which would be problematic outside of the changes made here to include it's usage in the release workflow. Is anything/anyone depending on that dockerfile to build opm in an overall multi-stage image?

Reply:
operator-sdk is effectively [dependent](https://github.com/operator-framework/operator-sdk/blob/1be35bedc4356d8a804ba8258e455cb9a3f4e419/internal/olm/operator/registry/index_image.go#L43) on it, but we're flexible.

Reply:
> problematic outside of the changes made here to include it's usage in the release workflow

Agreed. WDYT about moving/renaming this new `opm.Dockerfile` to something like `release/goreleaser.opm.Dockerfile` to make it more obvious what context this is intended to be used in.

Reply:
Moving this to a release directory and retaining the old Dockerfile sounds reasonable to me - as an aside, I wasn't able to immediately find any documentation around the various Dockerfile(s) that are housed in this repository, which is unfortunate if you don't have much historical context.